api_fhir/converters/patientConverter.py

```diff
@@ -0,0 +1,256 @@
+                code = maritalStatus.get("code")
+                if code == MARIAL_STATUS_MARRIED:
+                    imis_insuree.marital = "M"
+                elif code == MARIAL_STATUS_NEVER_MARRIED:
+                    imis_insuree.marital = "S"
+                elif code == MARIAL_STATUS_DIVORCED:
+                    imis_insuree.marital = "D"
+                elif code == MARIAL_STATUS_WIDOWED:
+                    imis_insuree.marital = "W"
+                elif code == MARIAL_STATUS_UNKNOWN:
+                    imis_insuree.marital = "N"
+
+    @classmethod
+    def build_fhir_telecom(cls, fhir_patient, imis_insuree):
+        telecom = []
+        if imis_insuree.phone is not None:
+            phone = cls.build_fhir_contact_point(imis_insuree.phone, ContactPointSystem.PHONE.value,
+                                             ContactPointUse.HOME.value)
+            telecom.append(phone.__dict__)
+        if imis_insuree.email is not None:
+            email = cls.build_fhir_contact_point(imis_insuree.email, ContactPointSystem.EMAIL.value,
+                                             ContactPointUse.HOME.value)
+            telecom.append(email.__dict__)
+        fhir_patient.telecom = telecom
+
+    @classmethod
+    def build_imis_contacts(cls, imis_insuree, fhir_patient):
+        if fhir_patient.get('telecom') is not None:
+            for contact_point in fhir_patient.get('telecom'):
+                if contact_point.get("system") == ContactPointSystem.PHONE.value:
+                    imis_insuree.phone = contact_point.get("value")
+                elif contact_point.get("system") == ContactPointSystem.EMAIL.value:
+                    imis_insuree.email = contact_point.get("value")
+
+    @classmethod
+    def build_fhir_addresses(cls, fhir_patient, imis_insuree):
+        addresses = []
+        if imis_insuree.current_address is not None:
+            current_address = cls.build_fhir_address(imis_insuree.current_address, AddressUse.HOME.value,
+                                                     AddressType.PHYSICAL.value)
+            addresses.append(current_address.__dict__)
+        if imis_insuree.geolocation is not None:
+            geolocation = cls.build_fhir_address(imis_insuree.geolocation, AddressUse.HOME.value,
+                                                 AddressType.BOTH.value)
+            addresses.append(geolocation.__dict__)
+        fhir_patient.address = addresses
+
+    @classmethod
+    def build_imis_addresses(cls, imis_insuree, fhir_patient):
+        if fhir_patient.get('address') is not None:
+            for address in fhir_patient.get('address'):
+                if address.get("type") == AddressType.PHYSICAL.value:
+                    imis_insuree.current_address = address.get("text")
+                elif address.get("type") == AddressType.BOTH.value:
+                    imis_insuree.geolocation = address.get("text")
+
+    class Meta:
+        app_label = 'api_fhir'
```

Comment:
this class looks quite big, is a possibilty to split it ?

Reply:
I thought about how can I make it more generic but so far I didn't have a better idea. If I will figure out some better approach I will change that in the future. If you will have some idea please let me know.